packages/protocol/contracts/governance/Governance.sol

```diff
@@ -510,13 +512,14 @@ contract Governance is IGovernance, Ownable, Initializable, UsingBondedDeposits,
     dequeueProposalsIfReady();
     Proposal storage proposal = proposals[proposalId];
     require(_proposalExists(proposal) && dequeued[index] == proposalId);
-    if (isDequeuedProposalExpired(proposalId)) {
-      deleteDequeuedProposal(proposalId, index);
+    if (isDequeuedProposalExpired(proposal)) {
+      deleteDequeuedProposal(proposal, proposalId, index);
       return false;
     }
-    ProposalStage stage = _getDequeuedProposalStage(proposal.timestamp);
+    ProposalStage stage = getDequeuedProposalStage(proposal.timestamp);
     require(msg.sender == approver && !proposal.approved && stage == ProposalStage.Approval);
     proposal.approved = true;
+    proposal.totalWeight = totalWeight();
```

Comment:
Note: this needs to be here so that `totalWeight` is set even if no votes are cast during Referendum. Should I add a comment explaining this?